src/ripple/beast/cxx17/type_traits.h

```diff
@@ -35,13 +35,76 @@ namespace std {
 
     #endif // ! __cpp_lib_void_t
 
+    #if ! __cpp_lib_common_type_t
```

Comment:
This breaks the macOS build as `__cpp_lib_common_type_t` doesn't exist in libc++.  So we end up getting two definitions of `common_type_t`.  Furthermore no feature-test macros are mandated in C++14 or C++17.  And `__cpp_lib_common_type_t` is not mandated in C++20.

Will gcc compile without this addition?

Reply:
can confirm - getting compile errors related to `common_type_t` on macos.

Reply:
@HowardHinnant @mellery451 I've rebased the branch onto 1.4.0-b1, which gives us C++17, and thus got rid of this definition plus the use of it in LoadFeeTrack.cpp. Let me know if this is still a problem, please.

Reply:
Looks good on macOS. 👍